### PR TITLE
Introduce vite

### PR DESCRIPTION
**Pull request by andrzejewsky**
closes: https://github.com/saleor/saleor-dashboard/issues/2557

A draft of migration to Vite. It significantly improvers the performance of hmr which boost the producttivity.

Things to test:
- sentry
- env variables values, if it's applied properly
- prod build & docker

The new/updated scripts:
`npm run dev` - for running dev mode
`npm run build` - for building 
`npm run preview` - for preview the built production version

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
MARKETPLACE_URL=https://apps.saleor.io
SALEOR_APPS_ENDPOINT=https://apps.saleor.io/api/saleor-apps

### Do you want to run more stable tests?
To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] giftCard
3. [ ] category
4. [ ] collection
5. [ ] attribute
6. [ ] productType
7. [ ] shipping
8. [ ] customer
9. [ ] permissions
10. [ ] menuNavigation
11. [ ] pages
12. [ ] sales
13. [ ] vouchers
14. [ ] homePage
15. [ ] login
16. [ ] orders
17. [ ] products
18. [ ] app
19. [ ] plugins
20. [ ] translations
21. [ ] navigation
22. [ ] variants
23. [ ] payments

CONTAINERS=1

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -85,7 +84,6 @@ const App: React.FC = () => (
           <DateProvider>
             <LocaleProvider>
               <MessageManagerProvider>
-                <ServiceWorker />
```
Comment (krzysztofzuraw) [inline src/index.tsx:88]: **question:** isn't `ServiceWorker` needed for informing user about new version of dashboard?

File: scripts/removeSourcemaps.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  When the sentry is enabled, remove sourcemaps
+*/
+
+if (
+  process.env.SENTRY_ORG &&
+  process.env.SENTRY_PROJECT &&
+  process.env.SENTRY_DSN &&
+  process.env.SENTRY_AUTH_TOKEN
+) {
+  rimraf("./build/**/*.js.map", () => console.log("Source maps removed."));
```
Comment (krzysztofzuraw) [inline scripts/removeSourcemaps.js:15]: **suggestion**: maybe we can document why we are removing `js.maps` when Sentry is turned on for the future reference - e.g in `docs/` folder in repo 🤔

Comment (andrzejewsky) [reply scripts/removeSourcemaps.js:15]: hmm I think this file is enough, but I will put more info there

File: src/index.tsx (hunk 1 above)
Comment (andrzejewsky) [reply src/index.tsx:88]: service worker was never been loaded, I want to create a new issue for that (enabling SW with vite)

Comment (andrzejewsky) [reply src/index.tsx:88]: but yeah... you are right about that component 👍

**Review by Droniu — DISMISSED** [review]
Very nice job, Christmas is early again this year 👏

**Review by szczecha — DISMISSED** [review]

**Review by krzysztofzuraw — APPROVED** [review]

**Review by Droniu — DISMISSED** [review]

**Review by orzechdev — DISMISSED** [review]

**Review by krzysztofzuraw — DISMISSED** [review]

**Review by krzysztofzuraw — APPROVED** [review]

**Review by Droniu — APPROVED** [review]

**Review by orzechdev — APPROVED** [review]

**Pull request merged**